File: tp2_python/nombre_mystere.py
import random

print()


def nombre_mystere():
    valeur_min=-100
    valeur_max=100
    nombre_machine=random.randrange(-100, 101)
    tentative=0
    
    print("Le nombre à trouver se trouve entre" ,valeur_min, "et",valeur_max)
    print("Vous avez droit à 10 tentatives")

    while tentative<10:
        try:
            nombre_joueur=int (input ("Entrez un nombre"))
        except ValueError:
            print("Vous avez entré une mauvaise valeur")
        if nombre_joueur>=valeur_min and nombre_joueur<=valeur_max:
            if nombre_joueur>nombre_machine:
                 tentative+=1
                 print("Trop grand")
            elif nombre_joueur<nombre_machine:
                 tentative+=1
                 print("Trop petit")
            else:
                tentative+=1
                print("Bravo, le nombre est bien" ,nombre_machine, "vous avez réussi en" ,tentative, "tentative(s)")
                break
        else:
            print("Erreur le nombre que vous avez rentré ne se trouvze pas dans l'intervalle qui vous a été mentionné")
            print("Le nombre à trouver était",nombre_machine)

nombre_mystere()

File: BonusC++/NombreMystere.cpp
#include"Header.h"
//#include <cstdlib>
#include <random>

//la fonction random 
random_device rd;
mt19937 gen(rd());

int random(int low, int high)
{
	uniform_int_distribution<> dist(low, high);
	return dist(gen);
}


int NombreMystere() {
	int NombreMachine =random(-100, 100);
	int NombreUtilisateur;
	int tentative = 0;
	const int VALEUR_MAX = 100;
	const int VALEUR_MIN = -100;
	cout << "NOMBRE MYSTERE" << endl << " Le nombre a trouver est entre " << VALEUR_MIN << " et " << VALEUR_MAX << endl << "Vous avez droit a 10 tentatives " << endl; 
	

	while (tentative < 10)
	{
		cout << " Entrez un nombre " << endl;
		cin >> NombreUtilisateur;

		if (NombreUtilisateur >= VALEUR_MIN && NombreUtilisateur <= VALEUR_MAX)
		{
			if(NombreUtilisateur > NombreMachine)
			{
				tentative++;
				cout << "Trop grand" << endl;
			}
			else if (NombreUtilisateur < NombreMachine)
			{
				tentative++;
				cout << "Trop petit" << endl;
			}
			else
			{
				tentative++;
				cout << "Bravo,le nombre a trouver est bien " << NombreMachine << " Vous avez gagne en " << tentative << " coup(s)" << endl;
				break;
			}

		}
		else
		{
			cout << "Erreur! " << NombreUtilisateur << " que vous avez entre ne se trouve pas dans l'intervalle mentionne plus haut" << endl;
		}
	}
	cout << "Le nombre a trouver etait " << NombreMachine << endl;
	return NombreMachine;
}